### ISSUE TITLE
Project Specs Format

### ISSUE THREAD

**Issue by orkunkilic** [bounswe/bounswe2023group8#25]
We should agree on some format to list the specs. It can be something like that.

- Operation
  - Type
  - Required Fields
  - Access
  
For example;
 
 - Register new user
   - POST
   - [name: str, email: str, password: str]
   - Everyone
   
 - Create new post
   - POST
   - [title: str, description: str, tags: Array<str>]
   - Users

Please comment on this issue and lets agree upon a format.

After this, we need to specify the database model and schema (User: {id, name, email, passwordEncrypted, interesAreas ...}). It should be a new issue.

**Comment by orkunkilic** [comment]
One side note: Type is not model like user or post. It is the request type that helps us identify operations. 
Create: POST
Read: GET
Update: PUT
Deletete: DELETE

[Reference](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods)

**Comment by EgemenKaplan** [comment]
I think this is few steps ahead of what we are currently supposed to do. We still need a contextual collection of all the requirements and the scenarios that are within the scope of our product. 
If we consider this first meeting as a meeting with our customers, we still need to make a draft and show what we understood to our customers. Then, they will go over what we have and give us a more accurate list of their demands and we will modify the requirements. This will go back and forth until both our customers and our team know what the product will be. We cannot decide on any of the technical issues until we know what we are doing.

**Comment by EgemenKaplan** [comment]
Gotcha! We could indeed decide on a spec format for now and start building them only after the project reaches a level of certainty. We will see what the future holds for us 😄.

**Comment by bakikucukcakiroglu** [comment]
Outdated.
